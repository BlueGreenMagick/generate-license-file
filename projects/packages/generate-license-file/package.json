{
  "name": "generate-license-file",
  "version": "0.0.0",
  "description": "Generates a text file containing all of the licences for your production dependencies",
  "main": "dist/main.js",
  "license": "ISC",
  "prettier": "@generate-license-file/prettier-config",
  "bin": {
    "generate-license-file": "bin/generate-license-file"
  },
  "scripts": {
    "build": "del-cli ./dist&&tsc",
    "build:watch": "del-cli ./dist&&tsc --watch",
    "test": "jest",
    "test:ci": "jest --ci --coverage"
  },
  "contributors": [
    {
      "name": "Toby Smith",
      "url": "https://github.com/tobysmith568"
    },
    {
      "name": "Toby Bessant",
      "url": "https://github.com/tobybessant"
    }
  ],
  "files": [
    "dist",
    "bin"
  ],
  "keywords": [
    "cli",
    "license",
    "licence",
    "third",
    "party",
    "file",
    "generate",
    "oss"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/TobyAndToby/generate-license-file.git"
  },
  "bugs": {
    "url": "https://github.com/TobyAndToby/generate-license-file/issues"
  },
  "homepage": "https://github.com/TobyAndToby/generate-license-file#readme",
  "dependencies": {
    "arg": "^5.0.0",
    "cli-spinners": "^2.6.0",
    "enquirer": "^2.3.6",
    "esm": "^3.2.25",
    "glob": "^8.0.0",
    "inquirer": "^9.0.0",
    "license-checker": "^25.0.1",
    "ora": "^5.4.1"
  },
  "devDependencies": {
    "@types/glob": "^7.1.4",
    "@types/inquirer": "^8.0.0",
    "@types/jest": "^28.0.0",
    "@types/license-checker": "^25.0.3",
    "del-cli": "^4.0.0"
  }
}
