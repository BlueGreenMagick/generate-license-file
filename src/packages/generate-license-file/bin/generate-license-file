#!/usr/bin/env node

require = require('esm')(module /*, options*/);
require('../src/lib/cli').main(process.argv);
