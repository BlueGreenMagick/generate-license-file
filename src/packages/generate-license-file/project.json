{
  "name": "generate-license-file",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/generate-license-file/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/generate-license-file",
        "main": "packages/generate-license-file/src/index.ts",
        "tsConfig": "packages/generate-license-file/tsconfig.lib.json",
        "assets": [
          "packages/generate-license-file/*.md",
          "packages/generate-license-file/bin/**",
          "LICENSE.md"
        ],
        "buildableProjectDepsInPackageJsonType": "dependencies"
      }
    },
    "version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/version.mjs generate-license-file {args.version}"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs generate-license-file {args.tag}"
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/generate-license-file/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/generate-license-file/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
