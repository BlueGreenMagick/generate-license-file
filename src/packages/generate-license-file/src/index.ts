/* istanbul ignore file */
export { generateLicenseFile } from "./lib/generateLicenseFile";
export { getLicenseFileText } from "./lib/getLicenseFileText";
export { getProjectLicenses } from "./lib/getProjectLicenses";
export { LineEnding } from "./lib/lineEndings";
export { ILicense } from "./lib/models/license";
