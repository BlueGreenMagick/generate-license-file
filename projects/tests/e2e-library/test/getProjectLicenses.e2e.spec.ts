import { describeEachTestPackage } from "@generate-license-file/e2e-shared";
import { getProjectLicenses } from "generate-license-file";

describe("getProjectLicenses", () => {
  describeEachTestPackage(packageJsonUnderTest => {
    it("should match snapshot", async () => {
      const result = await getProjectLicenses(packageJsonUnderTest);

      expect(result).toMatchSnapshot();
    });
  });
});
