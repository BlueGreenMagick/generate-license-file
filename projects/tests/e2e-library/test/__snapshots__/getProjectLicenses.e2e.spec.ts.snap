// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProjectLicenses for package project-1/package.json with a relative path should match snapshot 1`] = `
[
  License {
    "content": "# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.
",
    "dependencies": [
      "dep-five@1.0.0",
    ],
  },
  License {
    "content": "# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.",
    "dependencies": [
      "dep-four@1.0.0",
    ],
  },
  License {
    "content": "# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-one@1.0.0",
    ],
  },
  License {
    "content": "# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.",
    "dependencies": [
      "dep-three@1.0.0",
    ],
  },
  License {
    "content": "# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-two-duplicate@1.0.0",
      "dep-two@1.0.0",
    ],
  },
]
`;

exports[`getProjectLicenses for package project-1/package.json with an absolute path should match snapshot 1`] = `
[
  License {
    "content": "# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.
",
    "dependencies": [
      "dep-five@1.0.0",
    ],
  },
  License {
    "content": "# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.",
    "dependencies": [
      "dep-four@1.0.0",
    ],
  },
  License {
    "content": "# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-one@1.0.0",
    ],
  },
  License {
    "content": "# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.",
    "dependencies": [
      "dep-three@1.0.0",
    ],
  },
  License {
    "content": "# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-two-duplicate@1.0.0",
      "dep-two@1.0.0",
    ],
  },
]
`;
